pkg/resources/allocator.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2022 Intel Corp. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package resources
+
+import (
+	"sort"
+
+	pluginapi "k8s.io/kubelet/pkg/apis/deviceplugin/v1beta1"
+
+	"github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pkg/types"
+)
+
+// DeviceSet is to hold and manipulate a set of HostDevice
+type DeviceSet map[string]types.HostDevice
```

Comment:
id prefer this to be a package internal type (change to lowercase)

Reply:
I think we need this to be public, so that we can use `DeviceSet` in tests.

https://github.com/k8snetworkplumbingwg/sriov-network-device-plugin/pull/443/files#diff-8dc00b1b9c75fe62651b7f963c2fb2b56d0d0da00f2aa353c1e511cc1dea3151R79-R84